OptimoveCore/Sources/Classes/NetworkClient/NetworkClient.swift

```diff
@@ -4,6 +4,22 @@ import Foundation
 
 public typealias NetworkServiceCompletion = (Result<NetworkResponse<Data?>, NetworkError>) -> Void
 
+public extension NetworkClient {
+    @available(iOS 13.0, *)
+    func performAsync(_ request: NetworkRequest) async throws -> Data? {
+        return try await withCheckedThrowingContinuation { continuation in
+            perform(request) { result in
+                switch result {
+                case .success(let response):
+                    continuation.resume(returning: response.body)
+                case .failure(let error):
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+}
+
```

Comment:
I'm not entirely sure that this was the way to go, I was mostly trying to stop the errors from coming 😅 
What do you think about this @eligutovsky